src/macros.jl

```diff
@@ -39,10 +39,17 @@ function buildrefsets(expr::Expr)
         end
     end
     for s in c.args
-        if isa(s,Expr) && (s.head == :(=) || s.head == :in)
+        if isa(s, Expr) && ((s::Expr).head === :(=) || (s::Expr).head === :in)
+            s = s::Expr
             idxvar = s.args[1]
             idxset = esc(s.args[2])
-            push!(idxpairs, IndexPair(s.args[1],s.args[2]))
+            push!(idxpairs, IndexPair(s.args[1], s.args[2]))
+        elseif (isa(s, Expr) && (s::Expr).head === :comparison &&
```

Comment:
Better to wrap this in ``VERSION`` conditions so that we know to remove a branch when we drop support for 0.4